packages/web-features-figuring/src/browser-compat-data/supportStatements.ts

```diff
@@ -0,0 +1,180 @@
+    super(`${name} of ${value} is not a BCD real value`);
+  }
+}
+
+export class SupportStatement {
+  data: Partial<SimpleSupportStatement>;
+  browser: Browser | undefined;
+  feature: Feature | undefined;
+
+  constructor(
+    data: Partial<SimpleSupportStatement>,
+    browser?: Browser,
+    feature?: Feature,
+  ) {
+    this.data = data;
+    this.browser = browser;
+    this.feature = feature;
+  }
+
+  _isRanged(key: "version_added" | "version_removed" | undefined): boolean {
+    if (key === undefined) {
+      return (
+        this._isRanged("version_added") || this._isRanged("version_removed")
+      );
+    }
+
+    const version = this.data?.[key];
+
+    if (
+      typeof version === "boolean" ||
+      version === undefined ||
+      version === null
+    ) {
+      return false;
+    }
+
+    return version.includes("≤");
+  }
+
+  hasCaveats(): boolean {
+    return (
+      this.data.alternative_name !== undefined ||
+      this.flags.length > 0 ||
+      this.partial_implementation === true ||
+      this.data.prefix !== undefined
+    );
+  }
+
+  get flags(): FlagStatement[] {
+    return this.data?.flags ?? [];
+  }
+
+  get partial_implementation(): boolean {
+    // Strictness guarantee: unset partial_implementation returns false
+    return this.data?.partial_implementation ?? false;
+  }
+
+  get version_added(): VersionValue {
+    // Strictness guarantee: unset version_added returns false
+    return this.data?.version_added || false;
+  }
+
+  get version_removed(): VersionValue {
+    // Strictness guarantee: unset version_removed returns false
+    return this.data?.version_removed || false;
+  }
+}
+
+export class RealSupportStatement extends SupportStatement {
+  constructor(
+    data: Partial<SimpleSupportStatement>,
+    browser?: Browser,
+    feature?: Feature,
+  ) {
+    // Strictness guarantee: Support statements never contain non-real values
+
+    super(data, browser, feature);
+
+    if (!Object.hasOwn(data, "version_added")) {
+      throw new Error("version_added missing from simple support statement");
+    }
+
+    const { version_added } = data;
+    if (!(typeof version_added === "string" || version_added === false)) {
+      throw new NonRealValueError("version_added", version_added);
+    }
+
+    if (Object.hasOwn(data, "version_removed")) {
+      const { version_removed } = data;
+      if (!(typeof version_removed === "string" || version_removed === false)) {
+        throw new NonRealValueError("version_added", version_removed);
+      }
+    }
+  }
+
+  get version_added() {
+    return super.version_added as string | false;
+  }
+
+  get version_removed() {
+    return super.version_removed as string | false;
+  }
+
+  // TODO: `supportedBy()` ought to be (partially) implemented on non-real value
+  // support statements. For example, `"version_added": true` should allow for
+  // returning `[this.browser.current()]` at least.
+  supportedBy() {
```

Comment:
Should this and `hasCaveats` move into the baseline directory, to keep this directly as an unopinionated wrapper around BCD?